src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -518,25 +490,29 @@ public void Dispose()
         public event EventHandler<TxEventArgs>? NewPending;
         public event EventHandler<TxEventArgs>? RemovedPending;
 
-        private void Notify(ITxPoolPeer peer, Transaction tx, bool isPriority)
+        private void Notify(PeerInfo peer, Transaction tx, bool isPriority)
         {
             try
             {
-                peer.SendNewTransaction(tx, isPriority);
-                Metrics.PendingTransactionsSent++;
-                if (_logger.IsTrace) _logger.Trace($"Notified {peer.Enode} about a transaction: {tx.Hash}");
+                if (!peer.NotifiedTransactions.Get(tx.Hash))
+                {
+                    peer.NotifiedTransactions.Set(tx.Hash);
```

Comment:
better PeerInfo.SendTransaction and handle adding to cache internally (encapsulation)